test/libsolidity/semanticTests/various/value_complex.sol

```diff
@@ -14,7 +14,7 @@ contract test {
 
     function sendAmount(uint256 amount) public payable returns (uint256 bal) {
         uint256 someStackElement = 20;
-        return h.getBalance.value(amount).gas(1000).value(amount + 3)();
+        return h.getBalance{value: amount, gas: 1000}();
```

Comment:
Not sure if these changes should be made. Somehow in isoltest, I get the following message:

```
Exception during test: /solidity/libsolidity/interface/CompilerStack.cpp(648): Throw in function const solidity::evmasm::LinkerObject& solidity::frontend::CompilerStack::object(const string&) const
Dynamic exception type: boost::wrapexcept<solidity::langutil::CompilerError>
std::exception::what: Compilation was not successful.
[solidity::util::tag_comment*] = Compilation was not successful.
```

It also fails in the CI with the following error:

```
:18:16: Error: Duplicate option "value".
        return h.getBalance{value: amount, gas: 1000, value: amount + 3}();
               ^-------------------------------------------------------^
Warning: This is a pre-release compiler version, please do not use it in production.
:17:16: Error: Duplicate option "value".
        return h.getBalance{value: amount, gas: 1000, value: amount + 3}(); // overwrite value
               ^-------------------------------------------------------^
```

however, manually running it in solc, merely produces an error message (same as the CI.)

The same for the file value_insane.sol.

Checking for using value twice is already done in the file calloptions_duplicated.sol. Unless this code is supposed to serve a different purpose, it should be safe to make this change.

Reply:
This is what I'm getting currently from isoltest. @ekpyron can you have a look, because I don't understand this warning (the 5 -> 5 v/s 5 -> 8, I can fix.) 

```
semanticTests/various/value_insane.sol: FAIL
  Contract:
    contract helper {
        function getBalance() public payable returns (uint256 myBalance) {
            return address(this).balance;
        }
    }
    
    
    contract test {
        helper h;
    
        constructor() public payable {
            h = new helper();
        }
    
        function sendAmount(uint256 amount) public returns (uint256 bal) {
            return h.getBalance{value: amount, gas: 1000}();
        }
    }
    

  Expected result:
  // constructor(), 20 wei ->
  // sendAmount(uint256): 5 -> 8

  Obtained result:
  // constructor(), 20 wei ->
  // sendAmount(uint256): 5 -> 5
  Warning: The call to "sendAmount(uint256)" returned 
  [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,5]
  
  Attention: Updates on the test will apply the detected format displayed.
```

Reply:
Are you sure ``isoltest`` doesn't also first output the same error and only then tells you that it couldn't execute the test, because it couldn't compile it?
Apart from that: the original test had overwritten "amount" with "amount + 3", thus the expectation was "8" not "5". That's what the isoltest call is telling you: the previously expected return value is 8, the actual return value is 5. The "Warning: The call to... returned [0,...,0,5]" merely gives you the precise bytewise return data of the call (which is one 256-bit integer value 5).

I think you should do the changes in these tests - but for this one I would keep it at ``amount + 3``.

Reply:
Ping me in the chat, if this is still unclear :-).